File: Inventory-System/src/main/java/com/mceit_z/Inventory_System/mappers/rol/RolMapper.java
package com.mceit_z.Inventory_System.mappers.rol;

import com.mceit_z.Inventory_System.dto.rol.RolCreateDTO;
import com.mceit_z.Inventory_System.dto.rol.RolReadDTO;
import com.mceit_z.Inventory_System.dto.rol.RolUpdateDTO;
import com.mceit_z.Inventory_System.models.Rol;
import org.springframework.stereotype.Component;

@Component
public class RolMapper {

    public Rol toEntity(RolCreateDTO rolCreateDTO) {
        Rol rol = new Rol();
        rol.setName(rolCreateDTO.getName());
        return rol;
    }

    public RolReadDTO toReadDTO(Rol rol) {
        RolReadDTO dto = new RolReadDTO();
        dto.setIdRol(rol.getIdRol());
        dto.setName(rol.getName());
        return dto;
    }


    public void updateEntityFromDTO(RolUpdateDTO dto, Rol rol) {
        if (dto.getName() != null) {
            rol.setName(dto.getName());
        }
    }
}

File: Supermarket_Inventory_System_App/src/app/core/mappers/rol.mapper.ts
import { RolCreateDTO } from '../../infraestructure/dto/rol/rolCreate.dto';
import { RolReadDTO } from '../../infraestructure/dto/rol/rolRead.dto';
import { Rol } from '../models/rol.model';

export class RolMapper {
  static fromApiToDomain(apiRol: RolReadDTO): Rol {
    return {
      idRol: apiRol.idRol,
      name: apiRol.name,
    };
  }

  static fromDomainToApi(rol: Rol): RolCreateDTO {
    return {
      name: rol.name,
    };
  }
}
